### PR TITLE
Add image and polygons bounding box query

### PR DESCRIPTION
**Pull request by kevinyamauchi**
This PR adds bounding box queries for the image, labels and polygons

### CODE REVIEW

**Comment by kevinyamauchi** [comment]
It looks like the test failures are unrelated (they are writers). @LucaMarconato  or @giovp , have you seen this before?

**Comment by kevinyamauchi** [comment]
I have created an example of using the bounding box query on a SpatialData object containing images, points, and polygons here: https://gist.github.com/kevinyamauchi/29698d258077cf15c26c1ed042f81bb0

I think this is ready for review! The test failures appear unrelated. The pre-commit failure is not from code I touched. Should I re-run pre-commit on those files locally and push?

File: spatialdata/_core/_spatial_query.py
Comment (kevinyamauchi) [inline spatialdata/_core/_spatial_query.py:153]: should I use the set_transform convenience function here?

File: spatialdata/_core/_spatial_query.py
Comment (kevinyamauchi) [inline spatialdata/_core/_spatial_query.py:131]: should I use the get_transform convenience function here?

File: spatialdata/_core/_spatial_query.py
Comment (kevinyamauchi) [inline spatialdata/_core/_spatial_query.py:130]: this whole section will have to be refactored when #100 lands. However, in order to keep things moving, I would like to merge this and refactor later if possible.

File: spatialdata/_core/_spatial_query.py
Comment (giovp) [inline spatialdata/_core/_spatial_query.py:95]: with `__annotations__` we could replace Union with `FirstType | SecondType`. Just a comment, since if we settle with that we'd have to change it everywhere.

File: spatialdata/_core/_spatial_query.py
Comment (giovp) [inline spatialdata/_core/_spatial_query.py:108]: I don't think we use arguments in `Returns`

File: spatialdata/_core/_spatial_query.py
Comment (giovp) [reply spatialdata/_core/_spatial_query.py:153]: I think this will change also because we'll have a mapping in the `attrs["transform"]` so I think for now it's fine?

**Review by giovp — CHANGES_REQUESTED** [review]
really nice and clean code, left some comments re docstrings mostly for consistencies with rest of repo.

File: spatialdata/_core/_spatial_query.py
Comment (kevinyamauchi) [reply spatialdata/_core/_spatial_query.py:153]: sounds good. thanks!

File: spatialdata/_core/_spatial_query.py
Comment (kevinyamauchi) [reply spatialdata/_core/_spatial_query.py:101]: Okay, I will remove them.

File: spatialdata/_core/_spatial_query.py
Comment (kevinyamauchi) [reply spatialdata/_core/_spatial_query.py:108]: Should it be like this then?

```python
Returns
-------
The image contained within the specified bounding box.
```

Comment (giovp) [reply spatialdata/_core/_spatial_query.py:108]: Yes I think this is what we use

File: spatialdata/_core/_spatial_query.py
Comment (LucaMarconato) [reply spatialdata/_core/_spatial_query.py:130]: I agree with you

File: spatialdata/_core/_spatial_query.py
Comment (LucaMarconato) [reply spatialdata/_core/_spatial_query.py:95]: I am fine with that, maybe let's open an issue re refactoring them with that and once that pr is approved we can start using the new syntax.

**Comment by LucaMarconato** [comment]
@kevinyamauchi looks very good to me! I think there should be tests for labels since their axes is different from the image ones and since 3d labels are available. After that it's ready to merge!

**Comment by kevinyamauchi** [comment]
> @kevinyamauchi looks very good to me! I think there should be tests for labels since their axes is different from the image ones and since 3d labels are available. After that it's ready to merge!

Good point, @LucaMarconato . I can add this today.

**Comment by kevinyamauchi** [comment]
@giovp , @LucaMarconato , I have added the requested tests and fixed the docstrings!

**Comment by giovp** [comment]
thanks a lot @kevinyamauchi ! now that this is merged, could you by any chance take a look at the new points specs with dask dataframes https://github.com/scverse/spatialdata/pull/113 ? Thanks in advance.

**Comment by LucaMarconato** [comment]
@giovp I have fixed it (it was a minor thing)

**Pull request merged**
